app/controllers/cinema_halls_controller.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class CinemaHallsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :find_cinema_hall, only: %i[show update destroy]
```

Comment:
This is not wrong. But it's really hard to follow. So when I look for a show method I need to look above for a before_action and then to private to know what's going on. Sometimes DRY is not a best choice. https://blog.ragnarson.com/2015/09/30/problems-with-typical-rails-controllers-and-before-actions.html

Reply:
Is the below code something you have in mind? 🤔 

`# frozen_string_literal: true

class CinemaHallsController < ApplicationController
  before_action :authenticate_user!

  def index
    cinema_halls = CinemaHall.all

    render json: cinema_halls
  end

  def show
    find_cinema_hall
    render json: @cinema_hall
  end

  def create
    cinema_hall = CinemaHall.new(cinema_hall_params)
    if cinema_hall.save
      render json: cinema_hall, status: :created, location: cinema_hall
    else
      render json: cinema_hall.errors, status: :unprocessable_entity
    end
  end

  def update
    find_cinema_hall
    if @cinema_hall.update(cinema_hall_params)
      render json: @cinema_hall
    else
      render json: @cinema_hall.errors, status: :unprocessable_entity
    end
  end

  def destroy
    find_cinema_hall
    @cinema_hall.destroy
    head :no_content
  end

  private

  def find_cinema_hall
    @cinema_hall = CinemaHall.find(params[:id])
  end

  def cinema_hall_params
    params.require(:cinema_hall).permit(:name, :row_number, :row_total_seats)
  end
end`

Reply:
refactored, thanks :)